### PR TITLE
sig-release: Blockade changes to critical k/release tooling

### PR DESCRIPTION
**Pull request by justaugustus**
ref: https://github.com/kubernetes/release/issues/816
Signed-off-by: Stephen Augustus <saugustus@vmware.com>

### CODE REVIEW

**Comment by justaugustus** [comment]
/hold
(I think I have a few more patterns to add)

**Comment by justaugustus** [comment]
Done.
/hold cancel

**Comment by cblecker** [comment]
/hold

Why wouldn't we just do this with OWNERS files?

**Comment by justaugustus** [comment]
@cblecker -- This is essentially saying I want explicit approval from a repo admin (SIG Release Chair) before changing one of these tools. We have top-level approvers in k/release that I still want to be able to approve other changes without impact. 

As the tools are scattered across the repo, I need an extra gate to protect them and I heard that `required_reviewers` doesn't actually get respected?
Some of this could probably be obviated if all of the tools were in a single (or a few) subdirectories with `no_parent_owners: true`, but at this moment, I don't have a clear picture of what could break if I move things around.

**Comment by cjwagner** [comment]
Sounds like using `blockade` is most appropriate here, but just as an FYI Tide support blocking merge to entire branches or repos using a merge blocking issue: https://github.com/kubernetes/test-infra/blob/master/prow/cmd/tide/config.md#merge-blocker-issues
https://github.com/kubernetes/test-infra/blob/152915ad94b86eff2258201ad4460f2e2a3218a5/prow/config.yaml#L458

**Comment by justaugustus** [comment]
@cjwagner -- Whoa, that's cool. Thanks for the tip! Agreed that blockade might still be the right path here.

**Comment by fejta** [comment]
/unassign
/assign @cjwagner 

No concerns with the general intent

**Comment by cblecker** [comment]
I started writing something up and then apparently closed the tab 🤦‍♂ .

Basically, I still feel like this is an anti-pattern (blockading files, and then manually removing the label each time you need to change them). It provides a different contributing structure than normal, and requires manual intervention.

I could be convinced to be okay with this for a limited period of time to help stabilize the situation (30d?). Ultimately though, reorganizing files (either move the scripts all together, or move the *other stuff* all together) and then using OWNERS files should be the goal.

**Comment by justaugustus** [comment]
@cblecker -- yep, the intent for the blockade is definitely temporary! 
I want to rearrange the tools but I don't feel comfortable doing so until we can identify where they're used and have jobs to protect changes to them. Much of that work is already in flight. :)

**Comment by cblecker** [comment]
What's a reasonable time frame for this work?

**Comment by justaugustus** [comment]
Spoke to @cblecker on Slack. 
Setting a deadline for 8/8/19 and added a comment to remove the blockade afterwards.

**Comment by justaugustus** [comment]
Thanks again, Christoph!
/hold cancel

File: prow/plugins.yaml [hunk 1]
```diff
@@ -156,6 +156,17 @@ blockades:
...
+# TODO(justaugustus): Remove this blockade after identifying jobs where release tooling is being leveraged and
+#                     reorganizing k/release to be more restrictive about who can approve changes.
+#                     Deadline: 8/8/19
+#                     ref:
+#                       - https://github.com/kubernetes/test-infra/pull/13328
+#                       - https://github.com/kubernetes/release/issues/816
+- repos:
+  - kubernetes/release
+  blockregexps:
+  - ^anago|branchff|build\/|changelog-update|compile-release-tools|debian\/|find_green_build|gcb\/|gcbmgr|lib\/|prin|push-build.sh|release-notify|relnotes|rpm\/
```
Comment (BenTheElder) [inline prow/plugins.yaml]: why not .*? :P

**Pull request merged**